src/Swift.Bindings/src/Emitter/StringEmitter/Handler/PropertyHandler.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CodeDom.Compiler;
+
+namespace BindingsGeneration;
+
+/// <summary>
+/// Factory class for creating instances of PropertyHandler.
+/// </summary>
+public class PropertyHandlerFactory : IFactory<BaseDecl, IPropertyHandler>
+{
+    public bool Handles(BaseDecl decl)
+    {
+        return decl is PropertyDecl;
+    }
+
+    public IPropertyHandler Construct()
+    {
+        return new PropertyHandler();
+    }
+}
+
+/// <summary>
+/// Handler class for property declarations that generates the binding code for Swift properties.
+/// </summary> 
+public class PropertyHandler : BaseHandler, IPropertyHandler
+{
+    private readonly MethodHandler _methodHandler = new();
+
+    // Dictionary of Swift property names that need to be renamed in C#
+    // Temporary workaround for https://github.com/dotnet/runtimelab/issues/2997 to keep StoreKit tests passing
+    private static readonly Dictionary<string, string> PropertyNameMappings = new()
```

Comment:
This is an ugly workaround for https://github.com/dotnet/runtimelab/issues/2997
The problem was present before this PR - it was just surfaced by a computed property in StoreKit.

Fixing the name conflicts is a bigger thing - this is something to solve when we have proper c# translation.

Reply:
Consider using NameProvider for this where we could handle more cases until we refactor this.